eventservice/api/utils/group_auth.py

```diff
@@ -61,6 +61,17 @@ def get_group_memberships():
 
 # This util method checks if a user has admin permissions in a group event
 def check_group_event_admin_access(event, group_memberships):
+    if event and event.get('groupIds'):
+        if len(event.get('groupIds')) != len(group_memberships):
+            return False
+        for groupId in event.get('groupIds'):
+            found = False
+            for group_member in group_memberships:
+                if groupId == group_member.get('id') and group_member.get('role') == 'admin':
```

Comment:
Here, the check should be that the user is an admin of all the groups in the `groupIds` field. Based on the current logic, it looks like `found` will be set to true if one of the groups in the list matches.